analyzers/src/SonarAnalyzer.Common/Helpers/TypeHelper.cs

```diff
@@ -131,11 +131,11 @@ public static bool IsAny(this ITypeSymbol typeSymbol, ImmutableArray<KnownType>
 
     public static bool Implements(this ITypeSymbol typeSymbol, KnownType type) =>
         typeSymbol is { }
-        && typeSymbol.AllInterfaces.Any(symbol => symbol.ConstructedFrom.Is(type));
+        && typeSymbol.AllInterfaces.Any(x => x.ConstructedFrom.Is(type));
 
     private static bool Implements(this ITypeSymbol typeSymbol, ISymbol type) =>
         typeSymbol is { }
-        && typeSymbol.AllInterfaces.Any(symbol => symbol.ConstructedFrom.Equals(type));
+        && typeSymbol.AllInterfaces.Any(x => type.IsDefinition ? x.OriginalDefinition.Equals(type) : x.Equals(type));
```

Comment:
I think this can be written as:
` typeSymbol.AllInterfaces.Any(x => x.Equals(type));`
Practically when I run the tests with the change above the UTs stay green.
From my understanding `type.IsDefinition` should be false for interfaces/classes with generics (for which we check against specific types, not their generic form) - for the rest of the classes `x.Equals` should do the same job with `x.OriginalDefinition.Equals(type)`.

Am I missing something here?

Reply:
We checked it online - UTs will be added.

Reply:
The new UTs will demonstrate where it breaks without the `type.IsDefinision`